drivers/pps/sysfs.c

```diff
@@ -97,3 +98,28 @@ const struct attribute_group *pps_groups[] = {
 	&pps_group,
 	NULL,
 };
+
+/*
+ * Module parameters
+ */
+
+int pps_max_sources = 32;
+
+static int pps_max_sources_write(const char *val, const struct kernel_param *kp)
```

Comment:
why not use param_set_int here?

Or for that matter, make the parameter a uint and use param_get_uint/param_set_uint, since it's not meaningful to have a negative number of pps sources.

Reply:
When I looked at the function prototype for one of the places it was used, the function input was an `int`. Looking at the other 2 places though, that's not consistent and the other two accept `unsigned`. I'll switch to `uint`.

Reply:
That being said, I'm also realizing this is used to allocate a character device at module init. So I'll still need a function to handle reallocating when it changes which is quite a bit more complicated.

Reply:
Switched to use `module_param()` with a type of uint since the callback isn't needed.